packages/react/src/date-range-picker/utils.ts

```diff
@@ -0,0 +1,66 @@
+import { isBefore, subMonths, differenceInMonths, closestTo } from 'date-fns';
+
+// If the end date is before the start date, swap the end date with the start
+// This prevents the users from typing invalid date ranges
+export function ensureValidDateRange(dateRange: {
+	from: Date | undefined;
+	to: Date | undefined;
+}) {
+	const { to, from } = dateRange;
+
+	if (from && to && isBefore(to, from)) {
+		return { from: to, to: from };
+	}
+
+	return dateRange;
+}
+
+// The default calendar month is the first month to display when the date picker is opened
+export function getCalendarDefaultMonth(
+	inputMode: 'from' | 'to' | undefined,
+	valueAsDateOrUndefined: { from: Date | undefined; to: Date | undefined },
+	yearRange: { from: number; to: number } | undefined,
+	numberOfMonths: 1 | 2
+): Date | undefined {
+	const value = (() => {
+		// If the date picker has a value, open up the month of the date value
+		// Open up the 'from' date if the user has clicked the start date picker
+		if (inputMode === 'from' && valueAsDateOrUndefined.from) {
+			return valueAsDateOrUndefined.from;
+		}
+		// Open up the 'to' date if the user has clicked the end date picker
+		if (inputMode === 'to' && valueAsDateOrUndefined.to) {
+			return valueAsDateOrUndefined.to;
+		}
+		// If a `yearRange` prop has been set, use the closest day to todays date
+		if (yearRange) {
+			// Create a date on the first day of the range
+			const earliestDateInRange = new Date(yearRange.from, 0, 1);
+			// Create a date on the last day of the range
+			const lastDateInRange = new Date(yearRange.to, 11, 31);
+			// Use the closest day to todays date
+			return closestTo(new Date(), [earliestDateInRange, lastDateInRange]);
+		}
+		// Otherwise, returning undefined will fallback to the current month (react-day-picker behaviour)
+		return undefined;
+	})();
+
+	if (value && inputMode === 'to' && numberOfMonths === 2) {
+		const monthsRangeCount = (() => {
+			return valueAsDateOrUndefined?.from && valueAsDateOrUndefined?.to
+				? differenceInMonths(
+						valueAsDateOrUndefined.to,
+						valueAsDateOrUndefined.from
+				  )
+				: -1;
+		})();
+
+		// If there is only 1 month between the "from" and "to" dates, show the "to" month on the left side so it matches the "from" date
+		if (monthsRangeCount === 0) return value;
+
+		// When there are 2 months being displayed (i.e. on desktop), show the "to" month on the right side
+		return subMonths(value, 1);
```

Comment:
I would not expect to not see the start date when it can easily fit when opening the end date picker 
<img width="774" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/6fb3b7d5-75c2-4f55-b948-c60060b6ab0b">

Reply:
Yeah, it's weird. Here's opening the start picker:

<img width="797" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/ccc356d1-9788-405d-9a68-e6f7a253e0d7">


And here's the end with the same dates:

<img width="788" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/853552/2e3eba32-acab-4d9d-a0b5-4b2840661e8d">

Reply:
@stowball this should be fixed once its been redeployed.